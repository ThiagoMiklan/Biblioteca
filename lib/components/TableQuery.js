"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _react = _interopRequireDefault(require("react"));

var _type_validations = require("../tools/type_validations");

var props_obj = {
  itens: "array",
  header: "boolean",
  footer: "boolean",
  definition: "string"
};

var TableQuery = function TableQuery(props) {
  (0, _type_validations.validate)(props, props_obj, "TableQuery");
  return assembleTable(props);
};

function assembleTable(props) {
  // validate(props,props_obj);
  var definition = props.definition == undefined ? "table" : "table " + props.definition;

  var code = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null); // elimina vazio e nÃ£o array


  if (props.itens != undefined && Array.isArray(props.itens)) {
    if (props.itens.length > 0) {
      code = /*#__PURE__*/_react.default.createElement("table", {
        className: definition
      }, props.header == true && assembleHeader(props.itens), assembleBody(props.itens), props.footer == true && assembleFooter(props.itens));
    }
  }

  return code;
}

function assembleHeader(itens) {
  var code_header = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null);

  var itens_header = makeArray(itens);
  code_header = /*#__PURE__*/_react.default.createElement("thead", null, /*#__PURE__*/_react.default.createElement("tr", null, itens_header.map(function (item_header) {
    return assembleRow(item_header);
  })));
  return code_header;
}

function assembleFooter(itens) {
  var code_footer = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null);

  var itens_footer = makeArray(itens);
  code_footer = /*#__PURE__*/_react.default.createElement("tfoot", null, /*#__PURE__*/_react.default.createElement("tr", null, itens_footer.map(function (item_footer) {
    return assembleRow(item_footer);
  })));
  return code_footer;
}

function assembleBody(itens) {
  var code = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null);

  var getRow = function getRow(item) {
    var row = "";
    var values = Object.values(item);
    row = values.map(function (value) {
      return /*#__PURE__*/_react.default.createElement("td", null, value);
    });
    return row;
  };

  code = /*#__PURE__*/_react.default.createElement("tbody", null, itens.map(function (item) {
    return /*#__PURE__*/_react.default.createElement("tr", null, getRow(item));
  }));
  return code;
}

function makeArray(itens) {
  var allAttributes = itens.map(function (obj) {
    return Object.keys(obj);
  });
  var set = new Set();
  allAttributes.forEach(function (element) {
    element.map(function (item) {
      return set.add(item);
    });
  });
  return (0, _toConsumableArray2.default)(set);
}

function assembleRow(item) {
  return /*#__PURE__*/_react.default.createElement("th", null, item);
}

var _default = TableQuery;
exports.default = _default;