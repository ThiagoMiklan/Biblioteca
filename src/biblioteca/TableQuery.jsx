// @flow
import React from 'react';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import PropTypes from 'prop-types';
import checkerDefinition from '../tools/checker.js';


type Props = {
    itens: Array<Object>,
    header?: bool,
    footer?: bool,
    definition?: string,
}

const TableQuery = (props:Props)=>{
    checkerDefinition(props.definition,"Table");
    return assembleTable(props);
}

function assembleTable(props){
    var table_definition = (props.definition ==  undefined)? "": props.definition;
    var definition = classnames(getClassName("table "+table_definition,"Table"));
    var code = <></>;
    var x =  props.header;
    // elimina vazio e não array
    if(props.itens !=undefined && Array.isArray(props.itens)){
        if(props.itens.length > 0 ){
            code = <table className={definition}>
                    { props.header == true && assembleHeader(props.itens)}
                    {assembleBody(props.itens)}
                    {props.footer ==  true && assembleFooter(props.itens)}
                   
                </table>
        }
    }

    return code;
}

function assembleHeader(itens) {
    var code_header = <></>
    
    var itens_header = makeArray(itens);

    code_header = <thead>
            <tr>
                {itens_header.map(item_header => assembleRow(item_header))}
            </tr>
        </thead>
    
    return code_header;
}

function assembleFooter(itens) {
    var code_footer = <></>

    var itens_footer = makeArray(itens);
    
    code_footer = <tfoot>
            <tr>
                {itens_footer.map(item_footer => assembleRow(item_footer))}
            </tr>
        </tfoot>
    return code_footer;
}

function assembleBody(itens) {
    var code = <></>;
    var getRow = (item) => {
        var row = "";
        var values: Array<Object> = Object.values(item);
        row = values.map(value => <td>{value}</td>)
        return row;
    }
    code = <tbody>
            {itens.map(item => <tr>{getRow(item)}</tr>)}
    </tbody>
    
    return code;
}

function makeArray(itens){
    // intera sobre os itens da query e retira deles todos os atributos possíveis
    var allAttributes =  itens.map(obj => Object.keys(obj));
    // cria um set para possuir uma estrutura que mantenha a propriedade de singularidade
    var set = new Set();
    // intera sobre cada valor encontrado - Sendo uma estrutura de array de objetos
    allAttributes.forEach(element => {
        // intera sobre cada objeto pegando seus valores e adicionando ao set
        // que irá mantê-lo único
        element.map(item =>set.add(item));
    });
    
    return [...set];// converte o set em array novamente;
}

function assembleRow(item) {
    return <th>{item}</th>

}

TableQuery.propTypes ={
    itens: PropTypes.array,
    footer: PropTypes.bool,
    header: PropTypes.bool
}


export default TableQuery;
