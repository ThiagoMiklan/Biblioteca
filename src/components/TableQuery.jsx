// @flow
import React from 'react';
import PropTypes from 'prop-types';
import {validate} from '../tools/type_validations';

type Props = {
    itens: Array<Object>,
    header?: bool,
    footer?: bool,
    definition?: string,
}

const props_obj = {
    itens: "array",
    header: "boolean",
    footer: "boolean",
    definition: "string",
}

const TableQuery = (props:Props)=>{
    return assembleTable(props);
}

function assembleTable(props){
    validate(props,props_obj);
    var definition = (props.definition ==  undefined)? "table": "table " + props.definition;
    var code = <></>;
    // elimina vazio e não array
    if(props.itens !=undefined && Array.isArray(props.itens)){
        if(props.itens.length > 0 ){
            code = <table className={definition}>
                    { props.header == true && assembleHeader(props.itens)}
                    {assembleBody(props.itens)}
                    {props.footer ==  true && assembleFooter(props.itens)}
                   
                </table>
        }
    }

    return code;
}

function assembleHeader(itens) {
    var code_header = <></>
    
    var itens_header = makeArray(itens);

    code_header = <thead>
            <tr>
                {itens_header.map(item_header => assembleRow(item_header))}
            </tr>
        </thead>
    
    return code_header;
}

function assembleFooter(itens) {
    var code_footer = <></>

    var itens_footer = makeArray(itens);
    
    code_footer = <tfoot>
            <tr>
                {itens_footer.map(item_footer => assembleRow(item_footer))}
            </tr>
        </tfoot>
    return code_footer;
}

function assembleBody(itens) {
    var code = <></>;
    var getRow = (item) => {
        var row = "";
        var values: Array<Object> = Object.values(item);
        row = values.map(value => <td>{value}</td>)
        return row;
    }
    code = <tbody>
            {itens.map(item => <tr>{getRow(item)}</tr>)}
    </tbody>
    
    return code;
}

function makeArray(itens){
    // intera sobre os itens da query e retira deles todos os atributos possíveis
    var allAttributes =  itens.map(obj => Object.keys(obj));
    // cria um set para possuir uma estrutura que mantenha a propriedade de singularidade
    var set = new Set();
    // intera sobre cada valor encontrado - Sendo uma estrutura de array de objetos
    allAttributes.forEach(element => {
        // intera sobre cada objeto pegando seus valores e adicionando ao set
        // que irá mantê-lo único
        element.map(item =>set.add(item));
    });
    
    return [...set];// converte o set em array novamente;
}

function assembleRow(item) {
    return <th>{item}</th>

}

TableQuery.propTypes ={
    itens: PropTypes.array,
    footer: PropTypes.bool,
    header: PropTypes.bool
}


export default TableQuery;
