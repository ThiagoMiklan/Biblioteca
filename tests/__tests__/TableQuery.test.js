import React from 'react';
import TableQuery from '../../src/components/TableQuery';
import renderer from 'react-test-renderer';

var itens = [
    { nome: "Jackson", email: "jackson@arb.com.br", telefone: "123456" },
    { nome: "Marty", email: "marty@arb.com.br", telefone: "123456" },
    { nome: "Walter", email: "walter@arb.com.br", telefone: "123456" },
    { nome: "Antony", email: "antony@arb.com.br", telefone: "123456" }
]

test('value correctly', () => {
    const componentWithValue = renderer
        .create(
            <div className="container is-fluid">
                <TableQuery definition="is-fullwidth is-bordered" itens={itens} header={true} />
                <TableQuery definition="is-fullwidth is-striped" itens={itens} header={true} />
                <TableQuery definition="is-fullwidth is-narrow" itens={itens} header={true} />
                <TableQuery definition="is-fullwidth is-hoverable" itens={itens} header={true} />
                <TableQuery definition="is-fullwidth" itens={itens} header={true} footer={true} />
                <TableQuery definition="is-fullwidth" itens={itens} footer={true} />
                <TableQuery definition="is-bordered is-striped is-narrow is-hoverable is-fullwidth" itens={itens} />
                <br></br>
            </div>
        )
        .toJSON();
    // componente com valor
    expect(componentWithValue).toMatchSnapshot();

    
});

