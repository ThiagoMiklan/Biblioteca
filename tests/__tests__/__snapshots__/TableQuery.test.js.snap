// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`value correctly 1`] = `
<div
  className="container is-fluid"
>
  <table
    className="table is-fullwidth is-bordered"
  >
    <thead>
      <tr>
        <th>
          nome
        </th>
        <th>
          email
        </th>
        <th>
          telefone
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jackson
        </td>
        <td>
          jackson@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Marty
        </td>
        <td>
          marty@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Walter
        </td>
        <td>
          walter@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Antony
        </td>
        <td>
          antony@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="table is-fullwidth is-striped"
  >
    <thead>
      <tr>
        <th>
          nome
        </th>
        <th>
          email
        </th>
        <th>
          telefone
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jackson
        </td>
        <td>
          jackson@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Marty
        </td>
        <td>
          marty@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Walter
        </td>
        <td>
          walter@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Antony
        </td>
        <td>
          antony@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="table is-fullwidth is-narrow"
  >
    <thead>
      <tr>
        <th>
          nome
        </th>
        <th>
          email
        </th>
        <th>
          telefone
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jackson
        </td>
        <td>
          jackson@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Marty
        </td>
        <td>
          marty@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Walter
        </td>
        <td>
          walter@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Antony
        </td>
        <td>
          antony@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="table is-fullwidth is-hoverable"
  >
    <thead>
      <tr>
        <th>
          nome
        </th>
        <th>
          email
        </th>
        <th>
          telefone
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jackson
        </td>
        <td>
          jackson@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Marty
        </td>
        <td>
          marty@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Walter
        </td>
        <td>
          walter@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Antony
        </td>
        <td>
          antony@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="table is-fullwidth"
  >
    <thead>
      <tr>
        <th>
          nome
        </th>
        <th>
          email
        </th>
        <th>
          telefone
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jackson
        </td>
        <td>
          jackson@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Marty
        </td>
        <td>
          marty@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Walter
        </td>
        <td>
          walter@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Antony
        </td>
        <td>
          antony@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <th>
          nome
        </th>
        <th>
          email
        </th>
        <th>
          telefone
        </th>
      </tr>
    </tfoot>
  </table>
  <table
    className="table is-fullwidth"
  >
    <tbody>
      <tr>
        <td>
          Jackson
        </td>
        <td>
          jackson@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Marty
        </td>
        <td>
          marty@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Walter
        </td>
        <td>
          walter@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Antony
        </td>
        <td>
          antony@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <th>
          nome
        </th>
        <th>
          email
        </th>
        <th>
          telefone
        </th>
      </tr>
    </tfoot>
  </table>
  <table
    className="table is-bordered is-striped is-narrow is-hoverable is-fullwidth"
  >
    <tbody>
      <tr>
        <td>
          Jackson
        </td>
        <td>
          jackson@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Marty
        </td>
        <td>
          marty@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Walter
        </td>
        <td>
          walter@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
      <tr>
        <td>
          Antony
        </td>
        <td>
          antony@arb.com.br
        </td>
        <td>
          123456
        </td>
      </tr>
    </tbody>
  </table>
  <br />
</div>
`;
